### DISCUSSION TITLE
Feature Request: Type inference for generic instance creation ("Diamond" operator)

### DISCUSSION THREAD

**Discussion by Nufflee** [dotnet/csharplang#1755] [Language Ideas] [answered]
C# should infer types when creating instances of generic objects.

So, instead of typing:
```cs
List<Dictionary<Dictionary<string, int>, int> myList = new List<Dictionary<Dictionary<string, int>, int>(); 
```
we could type:
```cs
List<Dictionary<Dictionary<string, int>, int> myList = new List<>(); 
```

This improves code readability but unlike `var` keyword, it keeps the type "close" to the identifier. I'm not aware of any drawbacks to this.

**Comment by DarthVella** [comment]
I want to say there's a proposal that would allow
```csharp
List<Dictionary<Dictionary<string, int>, int> myList = new();
```
but my search-fu is failing me.

**Comment by Nufflee** [comment]
Oh, [here](https://github.com/dotnet/csharplang/blob/master/proposals/target-typed-new.md) it is, I wasn't aware of that. I think that's exactly this. Though that only talks about fields apparently. We would sure need this with properties and local variables too but I guess this isn't the best place to discuss that.

**Comment by jnm2** [comment, marked as answer]
The issue is https://github.com/dotnet/csharplang/issues/100 and would cover any place the `new` expression can be used.

**Comment by ufcpp** [comment]
dup of championed https://github.com/dotnet/csharplang/issues/1349

**Comment by Nufflee** [comment]
@ufcpp  Not really, this is full type inference vs partial that's explained in that issue but this seems to be a dupe anyways, right?

**Comment by Joe4evr** [comment]
Please close.

**Comment by LeonG-ZA** [comment]
This is one thing I don't like about Java
